src/DataType.cpp

```diff
@@ -42,6 +44,31 @@ std::string data_type_to_string(DataType dtype) {
     return str;
 }
 
+DataType string_to_data_type(const std::string& dtype) {
+
+    static std::map<std::string, DataType> type_map = {
+        {"bool", DataType::Bool},
+        {"char", DataType::Char},
+        {"float", DataType::Float},
+        {"single", DataType::Float},
+        {"double", DataType::Double},
+        {"int8", DataType::Int8},
+        {"int16", DataType::Int16},
+        {"int32", DataType::Int32},
+        {"int64", DataType::Int64},
+        {"uint8", DataType::UInt8},
+        {"uint16", DataType::UInt16},
+        {"uint32", DataType::UInt32},
+        {"uint64", DataType::UInt64},
+        {"string", DataType::String},
+        {"nothing", DataType::Nothing}
+    };
+
+    std::string dtype_l = dtype;
+    std::transform(dtype_l.begin(), dtype_l.end(), dtype_l.begin(), ::tolower);
+
+    return type_map[dtype];
```

Comment:
1. I guess this should be `return type_map[dtype_1]` and not `return type_map[dtype]`
2. If `dtype_1` is not a key in the map this actually inserts a new element [1], which is not intended.

[1] http://www.cplusplus.com/reference/map/map/operator[]